posthog/hogql/test/test_printer.py

```diff
@@ -2031,6 +2032,38 @@ def test_lookup_organic_medium_type(self):
             printed,
         )
 
+    def test_currency_conversion(self):
+        query = parse_select("select hogql_convertCurrency('USD', 'EUR', 100, toDate('2021-01-01'))")
+        printed = print_ast(
+            query,
+            HogQLContext(team_id=self.team.pk, enable_select_queries=True),
+            dialect="clickhouse",
+            settings=HogQLGlobalSettings(max_execution_time=10),
+        )
+        self.assertEqual(
+            (
+                f"SELECT multiplyDecimal(divideDecimal(100, dictGet({EXCHANGE_RATE_DICTIONARY_NAME}, 'rate', %(hogql_val_0)s, toDateOrNull(%(hogql_val_2)s), 0)), dictGet({EXCHANGE_RATE_DICTIONARY_NAME}, 'rate', %(hogql_val_1)s, toDateOrNull(%(hogql_val_2)s), 0)) "
+                "LIMIT 50000 SETTINGS readonly=2, max_execution_time=10, allow_experimental_object_type=1, format_csv_allow_double_quotes=0, max_ast_elements=4000000, max_expanded_ast_elements=4000000, max_bytes_before_external_group_by=0"
+            ),
+            printed,
+        )
+
+    def test_currency_conversion_without_date(self):
```

Comment:
Can we include some tests where the currency property is null / undefined / garbage?

It's fine if we don't do anything sensible if there's garbage, but would be good to at least have the test.

Reply:
Oh I'm being dumb, we're not actualy checking the result

Reply:
Can you add some tests to test_query where this function is called and we check the result?

Reply:
Yeah, sure, let me add that 🚧

I think the printer will just print it, though? We'd benefit from an actual test, but where would that go? I assume it might be a better fit once we're actually using this on Web Analytics

Reply:
Ah, I didn't know about `test_query.py`, let me see what I can do there

Reply:
It's there now, and I have one of the coolest tests ever :)